zerver/worker/register_push_device_to_bouncer.py

```diff
@@ -0,0 +1,117 @@
+import logging
+import random
+import time
+from typing import Any
+
+from django.conf import settings
+from typing_extensions import override
+
+from zerver.lib.exceptions import (
+    InvalidBouncerPublicKeyError,
+    PushRegistrationLivenessTimedOutError,
+)
+from zerver.lib.push_notifications import PUSH_REGISTRATION_LIVENESS_TIMEOUT
+from zerver.lib.remote_server import send_to_push_bouncer
+from zerver.models import PushDevice
+from zerver.models.users import UserProfile, get_user_profile_by_id
+from zerver.tornado.django_api import send_event_on_commit
+from zerver.worker.base import QueueProcessingWorker, assign_queue
+
+if settings.ZILENCER_ENABLED:
+    from zilencer.views import do_register_remote_push_device
+
+logger = logging.getLogger(__name__)
+
+
+@assign_queue("register_push_device_to_bouncer")
+class RegisterPushDeviceToBouncerWorker(QueueProcessingWorker):
+    @staticmethod
+    def handle_liveness_timedout(user_profile: UserProfile, push_account_id: int) -> None:
+        PushDevice.objects.filter(user=user_profile, push_account_id=push_account_id).delete()
+        event = dict(
+            type="push_account",
+            op="remove",
+            push_account_id=str(push_account_id),
+        )
+        send_event_on_commit(user_profile.realm, event, [user_profile.id])
+
+        logging.error(
+            "Push device registration for user_id=%s, push_account_id=%s timedout.",
+            user_profile.id,
+            push_account_id,
```

Comment:
I think these functions should go into a new `zerver/lib/push_notifications.py` -- generally we try to keep queue workers as lean as possible, since they're more annoying to unit test than a normal function.

Actually let's do a new `zerver/lib/push_registration.py`; we may move other code there later, but `push_notifications.py` is huge and this would be a nice collection of stuff to split out.